src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,169 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int currentSize;
+    private Node<T> first;
+    private Node<T> last;
+    private Node<T> newNode;
+    private Node<T> currentNode;
+
+    static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(Node<T> prev, T item, Node<T> next) {
+            this.item = item;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (currentSize == 0) {
+            linkFirst(value);
+        } else {
+            linkLast(value);
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > currentSize) {
+            throw new IndexOutOfBoundsException();
+        }
+        if (index == 0) {
+            linkFirst(value);
+        } else if (index == currentSize) {
+            linkLast(value);
+        } else {
+            linkMiddle(value, index);
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexValidation(index);
+        return findByIndex(index).item;
```

Comment:
looks like we always call `indexValidation` before `findByIndex`
so let's move `indexValidation` in `findByIndex`